service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractMapper.java

```diff
@@ -0,0 +1,86 @@
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.github.mustachejava.Mustache;
+
+@Component
+public class EhrExtractMapper {
+    private static final Mustache EHR_EXTRACT_TEMPLATE = TemplateUtils.loadTemplate("ehr_extract_template.mustache");
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
+        .appendPattern("yyyyMMddHHmmss")
+        .parseDefaulting(ChronoField.NANO_OF_DAY, 0)
+        .toFormatter();
+
+    private final FhirParseService fhirParseService;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+    private final TimestampService timestampService;
+
+    @Autowired
+    public EhrExtractMapper(FhirParseService fhirParseService,
+            RandomIdGeneratorService randomIdGeneratorService,
+            TimestampService timestampService) {
+        this.fhirParseService = fhirParseService;
+        this.randomIdGeneratorService = randomIdGeneratorService;
+        this.timestampService = timestampService;
+    }
+
+    public EhrExtractTemplateParameters mapJsonToEhrFhirExtractParams(GetGpcStructuredTaskDefinition getGpcStructuredTaskDefinition,
+            String json) {
+        Bundle bundle = fhirParseService.parseResource(json, Bundle.class);
+
+        return prepareEhrFhirExtractParamsFromFhirBundle(getGpcStructuredTaskDefinition, bundle);
+    }
+
+    public String mapEhrExtractToXml(EhrExtractTemplateParameters ehrExtractTemplateParameters) {
+        return TemplateUtils.fillTemplate(EHR_EXTRACT_TEMPLATE, ehrExtractTemplateParameters);
+    }
+
+    private EhrExtractTemplateParameters prepareEhrFhirExtractParamsFromFhirBundle(
+            GetGpcStructuredTaskDefinition getGpcStructuredTaskDefinition,
+            Bundle bundle) {
+        EhrExtractTemplateParameters ehrExtractTemplateParameters = new EhrExtractTemplateParameters();
+        ehrExtractTemplateParameters.setEhrFolderId(randomIdGeneratorService.createNewId());
+        ehrExtractTemplateParameters.setPatientId(getGpcStructuredTaskDefinition.getNhsNumber());
+        ehrExtractTemplateParameters.setRequestId(getGpcStructuredTaskDefinition.getRequestId());
+        ehrExtractTemplateParameters.setEhrExtractId(randomIdGeneratorService.createNewId());
+        ehrExtractTemplateParameters.setAuthorOdsCode(getGpcStructuredTaskDefinition.getFromOdsCode());
+        ehrExtractTemplateParameters.setDestinationOdsCode(getGpcStructuredTaskDefinition.getFromOdsCode());
```

Comment:
destination is fromOddCode. fromOdsCode refers to who sent the original EhrRequest. The "destination" of our reply is the original sender